bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/internal/location/BasicLocation.java

```diff
@@ -152,6 +164,11 @@ public synchronized boolean set(URL value, boolean lock, String lockFilePath) th
 		location = value;
 		if (property != null)
 			configValues.setConfiguration(property, location.toExternalForm());
+		if (serviceRegistration != null) {
```

Comment:
Should assign to a local variable first to avoid an NPE if another thread calls `setRegistration(null)` or the method `setRegistration` should check for null input and never null out the serviceRegistration.

Reply:
setRegistration is only called once after registration of the object so there should not be any threading issue here.

Reply:
I have made `setRegistration` synchronized as well so it should never change concurrently